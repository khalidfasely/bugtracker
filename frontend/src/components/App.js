function App() {
  return (
    <div className="App">
      <header>
        Hello world from React        
      </header>
    </div>
  );
}

export default App;
