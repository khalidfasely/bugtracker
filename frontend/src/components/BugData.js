const BugData = ({ bug }) => (
    <div>
        <p>{bug?.id}</p>
        <p>{bug?.user?.username}</p>
        <p>{bug?.title}</p>
        <p>{bug?.description}</p>
        <p>{bug?.time}</p>
        <p>{bug?.classification}</p>
        <p>{`${bug?.active}`}</p>
        <div>
            {bug?.admins?.map(admin => <span key={admin} >{admin} </span>)}
        </div>
        <br />
        <div>
            {bug?.users_with?.map(user => <span key={user} >{user} </span>)}
        </div>
    </div>
);

export default BugData;