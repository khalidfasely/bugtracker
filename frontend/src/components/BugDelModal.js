import Modal from "react-modal";

const BugDelModal = ({ delModalOpen, setDelModalOpen, deleteBug }) => (
    <Modal
        isOpen={delModalOpen}
        onRequestClose={() => setDelModalOpen(false)}
    >
        If you delete this bug there is no way to return it!
        <button onClick={deleteBug} data-testid='delete_button'>Delete</button>
    </Modal>
);

export default BugDelModal;