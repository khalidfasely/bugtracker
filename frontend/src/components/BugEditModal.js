import Modal from "react-modal";

import NewBug from "./NewBug";

const BugEditModal = ({
    editModalOpen, setEditModalOpen,
    users, bug
}) => (
    <Modal
        isOpen={editModalOpen}
        onRequestClose={() => setEditModalOpen(false)}
    >
        <NewBug users={users} isEdit={true} bug={bug} setEditModalOpen={setEditModalOpen} />
    </Modal>
);

export default BugEditModal;