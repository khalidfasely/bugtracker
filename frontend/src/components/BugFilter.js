import { connect } from "react-redux";
import { setTextFilter, sortByDefault, sortByPriority } from "../actions/bugFilter";

export const BugFilter = ({ filters, setTextFilter, sortByDefault, sortByPriority }) => {
    const onSortByChange = (e) => {
        if (e.target.value === 'priority') {
            sortByPriority();
            return;
        };
        sortByDefault();
    };

    return (
        <div className="bugs-filter">
            <div className="input-group__item">
                <input
                    data-testid='search_input'
                    type="text"
                    className="text-input"
                    placeholder="Search bugs"
                    value={filters.text}
                    maxLength={35}
                    onChange={(e) => setTextFilter(e.target.value)}
                />
            </div>
            <div className="input-group__item">
                <select
                    data-testid='select_input'
                    className="select"
                    value={filters.sortBy}
                    onChange={onSortByChange}
                >
                    <option value="">Date</option>
                    <option value="priority">Priority</option>
                </select>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => ({
    filters: state.bugFilter
});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByPriority: () => dispatch(sortByPriority()),
    sortByDefault: () => dispatch(sortByDefault())
});

export default connect(mapStateToProps, mapDispatchToProps)(BugFilter);