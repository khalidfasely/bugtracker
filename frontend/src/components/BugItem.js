import { Link } from "react-router-dom";
import { connect } from "react-redux";

import Comment from "./Comment";

const BugItem = ({uname, bug, comments}) => {
    console.log(bug, comments);
    return (
        <>
            {
                bug?.users_with?.includes(uname) ?
                <div>
                    **** bug <br /><h1>{bug.title}</h1>
                    **** comments {comments?.map(comment => <Comment comment={comment} />)}
                </div> :
                <div>Not able to see this bug! <Link to="/new">Create one!</Link></div>
            }
        </>
    );
}

const mapStateToProps = (state) => ({
    uname: state.auth.uname,
    bug: state.bugItem.bug,
    comments: state.bugItem.comments
});

export default connect(mapStateToProps)(BugItem);