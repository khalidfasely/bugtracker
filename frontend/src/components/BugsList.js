const BugsList = ({ bug }) => (
    <div className="bug-list-item">
        <div className="bug-list-item__title-cls">
            <h3>{bug?.title}</h3>
            {
                bug?.active ?
                <p className={bug?.classification}>{bug?.classification}</p> :
                <abbr title="Not Active Anymore">
                    <p className="not-active">N.A</p>
                </abbr>
            }
        </div>
        <p className="bug-list-item__description">{bug?.description.substring(0, 120)}...</p>
        <div className="bug-list-item__user-time">
            <p>{bug?.user?.username}</p>
            <p>{bug?.time}</p>
        </div>
        {/*<div>
            {bug?.admins?.map(admin => <span key={admin}>{admin} </span>)}
        </div>
        <br />
        <div>
            {bug?.users_with?.map(user => <span key={user}>{user} </span>)}
        </div>*/}
    </div>
);

export default BugsList;