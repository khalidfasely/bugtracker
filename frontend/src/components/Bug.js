import { useState, useEffect } from "react";
import { connect } from "react-redux";
import { useParams } from "react-router-dom";
import { startSetBug } from "../actions/bug";

import BugItem from './BugItem';

export const Bug = ({ startSetBug }) => {
    const params = useParams();

    const [ableToSee, setAbleToSee] = useState(false);
    const [notFound, setNotFound] = useState('');

    useEffect(() => {
        startSetBug(params.pid, params.bid)
        .then((result) => {
            setNotFound('');

            if (result.message) {
                setNotFound(result.message);
            };
        })
        .then(() => setAbleToSee(true))
        .catch(er => console.error(er));
    }, [params])


    if (notFound) {
        return <div>Bug Not Found! -- {notFound}</div>;
    };

    return (
        <div>
            {
                ableToSee ?
                <div><BugItem /></div> :
                <div>You're not able to see this bug</div>
            }
        </div>
    );
};

const mapDispatchToProps = (dispatch) => ({
    startSetBug: (pid, bid) => dispatch(startSetBug(pid, bid))
});

export default connect(undefined, mapDispatchToProps)(Bug);