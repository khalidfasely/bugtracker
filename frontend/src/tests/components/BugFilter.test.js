import { render, fireEvent, act } from "@testing-library/react";

import { BugFilter } from "../../components/BugFilter";

let props;

beforeEach(() => {
    props = {
        filters: {text: '', sort: ''},
        setTextFilter: jest.fn(),
        sortByDefault: jest.fn(),
        sortByPriority: jest.fn()
    };
});

test('Should render BugFilter component correctly', () => {
    const { asFragment } = render(<BugFilter {...props} />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should change search input value', () => {
    const { getByTestId } = render(<BugFilter {...props} />);
    const searchInputEl = getByTestId("search_input");

    fireEvent.change(searchInputEl, {
        target: {
            value: 'Username'
        }
    });

    //expect(props.setTextFilter).toHaveBeenCalledTimes(1);
    expect(props.setTextFilter).toHaveBeenLastCalledWith('Username');
});

test('Should sort by default and priority', () => {
    const { getByTestId } = render(<BugFilter {...props} />);
    const selectInputEl = getByTestId("select_input");

    fireEvent.change(selectInputEl, {
        target: {
            value: 'priority'
        }
    });

    expect(props.sortByPriority).toHaveBeenCalled();

    fireEvent.change(selectInputEl, {
        target: {
            value: ''
        }
    });

    expect(props.sortByDefault).toHaveBeenCalled();
});