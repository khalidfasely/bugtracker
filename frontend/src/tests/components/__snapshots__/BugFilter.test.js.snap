// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render BugFilter component correctly 1`] = `
<DocumentFragment>
  <div
    class="bugs-filter"
  >
    <div
      class="input-group__item"
    >
      <input
        class="text-input"
        data-testid="search_input"
        maxlength="35"
        placeholder="Search bugs"
        type="text"
        value=""
      />
    </div>
    <div
      class="input-group__item"
    >
      <select
        class="select"
        data-testid="select_input"
      >
        <option
          value=""
        >
          Date
        </option>
        <option
          value="priority"
        >
          Priority
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
